software/base/src/main/java/org/apache/brooklyn/entity/software/base/InboundPortsUtils.java

```diff
@@ -77,12 +82,17 @@
             Set<ConfigKey<?>> configKeys = Sets.newHashSet(extraConfigKeys);
             configKeys.addAll(entity.getEntityType().getConfigKeys());
             // Also add dynamically added config keys
-            configKeys.addAll(((EntityInternal)entity).config().getBag().getAllConfigAsConfigKeyMap().keySet());
+            Map<ConfigKey<?>, ?> configuredConfigKeys = ImmutableMap.copyOf(((EntityInternal) entity).config().getBag().getAllConfigAsConfigKeyMap());
+            configKeys.addAll(configuredConfigKeys.keySet());
```

Comment:
@aledsage, thanks a lot for your explanation. 
I see the problem about `ImmutableMap` now and I understand the difference between both of them, so I will use `MutableMap`.
About the exception, the method terminated propagating a `NullPointerException` and I saw a `NullPointerException` in the console, but I did not understand why a `NullPointerException` was thrown. Then, I found an `InterruptedException` in the stack. I do not know if it makes sense for you, but this `InterruptedException` helped me to understand the problem and then I decided to save a copy of the information that was being accessed.